src/update_engine/delta_performer.cc

```diff
@@ -461,10 +464,18 @@ bool DeltaPerformer::PerformReplaceOperation(
   // point to one of the ExtentWriter objects above.
   ExtentWriter* writer = NULL;
   if (operation.type() == DeltaArchiveManifest_InstallOperation_Type_REPLACE) {
-    writer = &zero_pad_writer;
+    if (is_kernel_partition) {
```

Comment:
src/update_engine/delta_performer.cc:465:56: error: conditional expression between distinct pointer types ‘chromeos_update_engine::DirectExtentWriter*’ and ‘chromeos_update_engine::ZeroPadExtentWriter*’ lacks a cast
   ExtentWriter* writer = is_kernel ? &direct_writer : &zero_pad_writer;

I don't know that casting is cleaner than the original code here, and it takes us further away from upstream.